diff --git a/src/api/wayfire/view-transform.hpp b/src/api/wayfire/view-transform.hpp
index 626923a4..e8dc5903 100644
--- a/src/api/wayfire/view-transform.hpp
+++ b/src/api/wayfire/view-transform.hpp
@@ -99,8 +99,8 @@ class transformer_render_instance_t : public render_instance_t
         }
 
         auto bbox = self->get_children_bounding_box();
-        int target_width  = scale * bbox.width;
-        int target_height = scale * bbox.height;
+        int target_width  = bbox.width;
+        int target_height = bbox.height;
 
         OpenGL::render_begin();
         if (inner_content.allocate(target_width, target_height))

