diff --git a/src/core/seat/pointer.cpp b/src/core/seat/pointer.cpp
index ae0cda5f..8c4e176a 100644
--- a/src/core/seat/pointer.cpp
+++ b/src/core/seat/pointer.cpp
@@ -121,7 +121,7 @@ void wf::pointer_t::transfer_grab(scene::node_ptr node, bool retain_pressed_stat
     }
 
     LOGD("transfer grab ", cursor_focus.get(), " -> ", node.get());
-    force_release_buttons();
+
     cursor_focus = node;
 
     // Send pointer_enter to the grab
