diff --git client/CMakeLists.txt client/CMakeLists.txt
index 30bd6b85..442a6655 100644
--- client/CMakeLists.txt
+++ client/CMakeLists.txt
@@ -26,6 +26,7 @@ include(OptimizeForNative) # option(OPTIMIZE_FOR_NATIVE)
 include(UninstallTarget)
 
 find_package(PkgConfig)
+find_package(NanoSVG REQUIRED)
 pkg_check_modules(FONTCONFIG REQUIRED IMPORTED_TARGET fontconfig)
 pkg_check_modules(FUSE3 REQUIRED IMPORTED_TARGET fuse3>=3.10)
 
@@ -158,7 +159,6 @@ include_directories(
   ${PROJECT_SOURCE_DIR}/include
   ${CMAKE_BINARY_DIR}
   ${CMAKE_BINARY_DIR}/include
-  ${PROJECT_TOP}/repos/nanosvg/src
 )
 
 link_libraries(
@@ -255,6 +255,7 @@ target_link_libraries(looking-glass-client
   ${EXE_FLAGS}
   PkgConfig::FONTCONFIG
   PkgConfig::FUSE3
+  NanoSVG::nanosvg
   lg_resources
   lg_common
   displayservers
